packages/react/src/hooks/useMediaQuery.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useRef, useState } from 'react';
+
+const breakpointValues = {
+  xs: 320,
+  s: 576,
+  m: 768,
+  l: 992,
+  xl: 1248,
+};
+
+const getWindowInnerWidth: () => undefined | number = () =>
+  typeof window !== 'undefined' && window.innerWidth ? window.innerWidth : undefined;
+const isEqualOrGreaterThan = (a: number, b: number | undefined) => (typeof b !== 'undefined' && b >= a) || false;
+const isLesserThan = (a: number, b: number | undefined) => (typeof b !== 'undefined' && b < a) || false;
+
+/**
+ * Hook for checking if a certain viewport breakpoint has been met.
+ *
+ * @param {() => boolean} breakpointCheck
+ * @returns
+ */
+const useMediaQuery = (breakpointCheck: () => boolean): boolean => {
+  const [matches, _setIfMatches] = useState<boolean>(false);
+  /* Use state ref to access current value inside event listener function. */
+  const matchesRef = useRef(matches);
+
+  const setIfMatches = (val: boolean) => {
```

Comment:
Could this be renamed rather than "overwrite" the original hook set function?
`UpdateMatch`? `SetMatches`? `refreshIfMatches`?